core/chains/evm/client/node.go

```diff
@@ -452,18 +467,50 @@ func (n *node) HeaderByNumber(ctx context.Context, number *big.Int) (header *typ
 	return
 }
 
-func (n *node) SendTransaction(ctx context.Context, tx *types.Transaction) error {
+func (n *node) SendTransaction(ctx context.Context, tx *types.Transaction) (err error) {
+	lggr := n.newRqLggr(switching(n)).With("tx", tx)
+
+	var rpcDomain string
+	defer func(start time.Time) {
+		duration := time.Since(start)
+
+		// In the event where the remote node is down, we will return before
+		// we have a chance to set rpcDomain to something useful.
+		if rpcDomain == "" {
+			rpcDomain = "err-no-domain"
+		}
+
+		promEVMPoolSendTransactionTiming.
+			WithLabelValues(
+				n.chainID.String(),             // chain ID
+				n.name,                         // node name
+				rpcDomain,                      // rpc domain
+				"false",                        // is send only
+				strconv.FormatBool(err == nil), // is successful
+			).
+			Observe(float64(duration))
+
+		lggr.Debugw("RPC call: evmclient.Client#SendTransaction",
```

Comment:
Can we add timing info to logs for all RPC calls?

Reply:
Can do - do you think it should be one prometheus metric with a `"rpcCall"` label that is different for each RPC call? Or a different prom metric for each rpc call?

Reply:
nice that's going to be super useful

Reply:
+2 on single prometheus metric with different labels